### PR TITLE
Jjwt

### PR DESCRIPTION
**Pull request by amit2103**
New pull request for #2100 , the other one seems to be very outdated

### CODE REVIEW

**Comment by amit2103** [comment]
Havent deleted all XAUTH yet..once approved will do that

File: app/templates/src/main/java/package/config/_SecurityConfiguration.java [hunk 1]
```diff
@@ -84,8 +107,8 @@ public void configure(WebSecurity web) throws Exception {
             .antMatchers("/api/account/reset_password/init")
             .antMatchers("/api/account/reset_password/finish")<% } %>
             .antMatchers("/test/**")<% if (devDatabaseType != 'h2Disk' && devDatabaseType != 'h2Memory') { %>;<% } else { %>
-            .antMatchers("/h2-console/**");<% } %>
-    }<% if (authenticationType == 'session' || authenticationType == 'xauth') { %>
+            .antMatchers("/h2-console/**"");<% } %>
```
Comment (jdubois) [inline app/templates/src/main/java/package/config/_SecurityConfiguration.java]: looks like you have an extra quote here

File: app/index.js [hunk 2]
```diff
@@ -740,6 +761,73 @@ module.exports = JhipsterGenerator.extend({
...
+    if (this.devDatabaseType != "h2Disk" && this.devDatabaseType != "h2Memory" && this.devDatabaseType != "oracle") {
+        this.template('_docker-compose.yml', 'docker-compose.yml', this, {});
+    }
+    if (this.prodDatabaseType != "oracle" || this.searchEngine == "elasticsearch") {
+        this.template('_docker-compose-prod.yml', 'docker-compose-prod.yml', this, {});
+    }
+    if (this.devDatabaseType == "cassandra") {
+        this.template('_Dockerfile_cassandra', 'Dockerfile', this, {});
+    }
+>>>>>>> Removed unnecessary file
```
Comment (jdubois) [inline app/index.js]: you have a merge issue here

File: app/index.js [hunk 3]
```diff
@@ -740,6 +761,73 @@ module.exports = JhipsterGenerator.extend({
             this.template(RESOURCE_DIR + '/config/_application-dev.yml', RESOURCE_DIR + 'config/application-dev.yml', this, {});
             this.template(RESOURCE_DIR + '/config/_application-prod.yml', RESOURCE_DIR + 'config/application-prod.yml', this, {});
 
+            cb();
+        }.bind(this));
+    }
+};
+
+JhipsterGenerator.prototype.app = function app() {
```
Comment (jdubois) [inline app/index.js]: I don't think this should be part of your PR

**Comment by jdubois** [comment]
@amit2103 I corrected some errors, as well as the merge issues, but there are still a few of your classes missing, like `UserAuthentication` or `CustomUserDetails`. Can you check? And add those files if possible?
I am working on this branch: https://github.com/jhipster/generator-jhipster/tree/amit2103-jjwt

File: app/index.js [hunk 4]
```diff
@@ -859,6 +947,11 @@ module.exports = JhipsterGenerator.extend({
                 this.template('src/main/java/package/config/liquibase/_package-info.java', javaDir + 'config/liquibase/package-info.java', this, {});
             }
 
+    if (this.authenticationType == 'jwt') {
```
Comment (deepu105) [inline app/index.js]: This seems odd as well

File: app/index.js [hunk 5]
```diff
@@ -1001,8 +1094,41 @@ module.exports = JhipsterGenerator.extend({
...
+    this.template('src/main/java/package/security/_UserNotActivatedException.java', javaDir + 'security/UserNotActivatedException.java', this, {});
+
+     if (this.authenticationType == 'jwt') {
+        this.template('src/main/java/package/security/jwt/_TokenAuthenticationService.java', javaDir + 'security/jwt/TokenAuthenticationService.java', this, {});
+        this.template('src/main/java/package/security/jwt/_TokenHandler.java', javaDir + 'security/jwt/TokenHandler.java', this, {});
+        this.template('src/main/java/package/web/rest/_UserJWTTokenController.java', javaDir + 'web/rest/UserJWTTokenController.java', this, {});
+        this.template('src/main/java/package/security/jwt/_JWTTokenConfigurer.java', javaDir + 'security/jwt/JWTTokenConfigurer.java', this, {});
+        this.template('src/main/java/package/security/jwt/_StatelessAuthenticationFilter.java', javaDir + 'security/jwt/StatelessAuthenticationFilter.java', this, {});
+    }
+>>>>>>> Initial commit
```
Comment (deepu105) [inline app/index.js]: Here as well

File: app/index.js [hunk 6]
```diff
@@ -1162,10 +1288,17 @@ module.exports = JhipsterGenerator.extend({
                 this.copyJs(WEBAPP_DIR + '/scripts/app/account/social/_social-register.js', WEBAPP_DIR + 'scripts/app/account/social/social-register.js', this, {});
             }
 
+
             // interceptor code
             this.template(WEBAPP_DIR + '/scripts/components/interceptor/_auth.interceptor.js', WEBAPP_DIR + 'scripts/components/interceptor/auth.interceptor.js', this, {});
             this.template(WEBAPP_DIR + '/scripts/components/interceptor/_errorhandler.interceptor.js', WEBAPP_DIR + 'scripts/components/interceptor/errorhandler.interceptor.js', this, {});
             this.template(WEBAPP_DIR + '/scripts/components/interceptor/_notification.interceptor.js', WEBAPP_DIR + 'scripts/components/interceptor/notification.interceptor.js', this, {});
+    this.template('src/main/java/package/web/filter/_package-info.java', javaDir + 'web/filter/package-info.java', this, {});
```
Comment (deepu105) [inline app/index.js]: This shouldn't be here

**Comment by amit2103** [comment]
@jdubois  I will add those files..missed them...Will do it by tonight surely

**Comment by jdubois** [comment]
I'm sorry @amit2103 but your code does not compile, I've spent a couple of hours on it and there are just too many errors, I can't merge this.

**Comment by amit2103** [comment]
@jdubois , I will look into this and fix this by tomorrow. I found the problem,will fix and send a pull request to your new branch

**Comment by amit2103** [comment]
Right now I am in my day job, can you post the compilation errors by the by when you have time? I intend to go home and work on this today. Sorry to waste your time

**Comment by jdubois** [comment]
I'm currently starting again from scratch, I'm trying to understand the issues, I'll keep you posted

**Comment by amit2103** [comment]
Okay..If you find new comopilation errors keep me in the loop...also I put a new commit in and a pull request to your working branch..some time ago

**Comment by jdubois** [comment]
Thanks @amit2103 I'm starting to having it work -> I recoded everything, but I wouldn't have done it so quickly without your help, so thanks a lot!!!!!!!!! And besides you are the one who pushed for this idea, and this was a really good one!
I'll commit this soon, I really have strong expectations for this!

**Comment by amit2103** [comment]
@jdubois  No problems...Let me know if you need any help

**Comment by jdubois** [comment]
Well, now front-end tests are not running, so help will be needed there...(running "gulp test")

**Comment by amit2103** [comment]
Okay..are the X-AUTH related changes

**Pull request closed without merging**